drivers/sensor/amd_sb_tsi/sb_tsi.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT amd_sb_tsi
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/pm/device_runtime.h>
+#include <zephyr/logging/log.h>
+#include "sb_tsi.h"
+
+LOG_MODULE_REGISTER(AMD_SB_TSI, CONFIG_SENSOR_LOG_LEVEL);
+
+static int sb_tsi_sample_fetch(const struct device *dev,
+			       enum sensor_channel chan)
+{
+	struct sb_tsi_data *data = dev->data;
+	const struct sb_tsi_config *config = dev->config;
+	int res;
+
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_DIE_TEMP) {
+		return -ENOTSUP;
+	}
+
+	res = pm_device_runtime_get(dev);
+	if (res) {
+		return res;
+	}
```

Comment:
I don't see other sensors using power management this way.  There are few examples of sensors using the power management action to put the sensor into a low power mode.  But what you've implemented here would prevent a power domain from going into suspend.
[bmi323 example](https://github.com/zephyrproject-rtos/zephyr/blob/762a1f8b42412c24f76b1533df69fc5d27b8b70a/drivers/sensor/bmi323/bmi323.c#L1235)

@teburd - you mentioned defining a PM strategy for sensors [here](https://github.com/zephyrproject-rtos/zephyr/pull/55790#issuecomment-1472001677).  What do you think?

Reply:
I changed it to be like how it's done in bmi160 (as linked above).